### PR TITLE
Add support for Hetzner DNS

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Add support for Hetzner DNS, based on https://github.com/markuman/hetzner_dns.

The modules are minimally adjusted copies of the `hosttech_*` versions. And some other preparations were already done in #26.

CC @markuman

##### ISSUE TYPE
- Feature Pull Request
- New Module Pull Request

##### COMPONENT NAME
Hetzner DNS module utils and modules

### CODE REVIEW

File: plugins/module_utils/hetzner/api.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        """
+        if record.id is None:
+            raise DNSAPIError('Need record ID to delete record!')
+        dummy, info = self._delete('v1/records/{id}'.format(id=record.id), must_have_content=False, expected=[200, 404])
+        return info['status'] == 200
+
+
+def create_hetzner_argument_spec():
+    return ArgumentSpec(
+        argument_spec=dict(
```
Comment (markuman) [inline plugins/module_utils/hetzner/api.py:189]: @felixfontein can we add alias values for `prefix` (_name_)  and `zone` (_zone_name`)?  
That would make `community.dns` become a drop-in replacement.

Comment (felixfontein) [reply plugins/module_utils/hetzner/api.py:189]: There are some more differences:
1. `state` has no default;
2. the interface is currently quite similar to `route53`, so `overwrite: true` has be added in quite a few cases.

I'm wondering whether the interface of the `_record` module shouldn't be overhauled anyway. Maybe let's discuss this in a separate issue; including which aliases to offer (in deprecated form for migration, and potentially permanently?).

**Comment by felixfontein** [comment]
I think this is a good first version. My next steps would be (once this is merged):
1. Look at #29.
2. Refactor the code so that `fetch_url` / `AnsibleModule` usage can be replaced by something that works with plugins, so we can add an inventory module similar to https://github.com/markuman/hetzner_dns/blob/main/plugins/inventory/inventory.py

**Comment by felixfontein** [comment]
While testing this, I noticed a problem with idempotency when using TXT records: Hetzner's API accepts unquoted values and reformats them with quotes (to conform to the 256 character limit, see https://datatracker.ietf.org/doc/html/rfc1035#section-3.3, https://datatracker.ietf.org/doc/html/rfc1035#section-3.3.14), and returns them in quoted form. The module right now does not know about this, which causes idempotency problems. For example when I try to set a DKIM entry
```
v=DKIM1; k=rsa; p=MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAw5AwkgAWIK9IANXXo1Pzhk2F3aYjTETQ9pJvIHnPhKS7vK3dBtaox0oU7o481/v3cHzgBVfsubDcNHbUPmvR84ouNXPowLbUkXwRIpSEZfUPuf6LBfFL1tFXBl7xtCRomr0VGAACHUK4L8mmWoP1Ah3iH54163oncG20JI8xfJjHfKAscVD3k3Ls3gU/2J33gptHCZunXlW1beFb/OiDl6ECGB2KhYwSWN/AIsO8VfVIRpCRswK1WcTgN3ZwoiLIktBEJicktlPB6HU8PhRl/9u0O3pkNB/pFMx0iViNEVEH4Sr4v4urLOqeE2LYo0TVPDn1wtzOwOICFFxtVYxY0zO/NtwcYDJJNbRVFSBKP1ePhf7uFo2HS8GamVdeyBi/+vRbcGZ1/QMTRR82aPNnyx6azCEXjdAcGPBV4pRPjUfbS1sZzhNiUEGYqGWypZtaLbJ2yBDOXfCLpaaf1VctqJ93KhONDLPnM2pREK48/fM0mn6x9uuCSJXo3FSW9xjnUXzj7cGxOvtdX2qnfTitiIiPIZNb4k6qbBbft1rI8oHYnt5+YUeVcrsSpoLL2QNAx9bavhv/hTO6Av/7Qofw/caGcSpx9kPGaxMjSBIpa32zrwRj/0qDVm7xgp1qkZayv54URd/nyqrNx0ZC8ZxS2mKLGWiL5bsOkNcEGe0lzeUCAwEAAQ==
```
Either this way or with double quotes, the API always returns the value as
```
"v=DKIM1; k=rsa; p=MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAw5AwkgAWIK9IANXXo1Pzhk2F3aYjTETQ9pJvIHnPhKS7vK3dBtaox0oU7o481/v3cHzgBVfsubDcNHbUPmvR84ouNXPowLbUkXwRIpSEZfUPuf6LBfFL1tFXBl7xtCRomr0VGAACHUK4L8mmWoP1Ah3iH54163oncG20JI8xfJjHfKAscVD3k3Ls3gU/2J33g" "ptHCZunXlW1beFb/OiDl6ECGB2KhYwSWN/AIsO8VfVIRpCRswK1WcTgN3ZwoiLIktBEJicktlPB6HU8PhRl/9u0O3pkNB/pFMx0iViNEVEH4Sr4v4urLOqeE2LYo0TVPDn1wtzOwOICFFxtVYxY0zO/NtwcYDJJNbRVFSBKP1ePhf7uFo2HS8GamVdeyBi/+vRbcGZ1/QMTRR82aPNnyx6azCEXjdAcGPBV4pRPjUfbS1sZzhNiUEGYqGWypZta" "LbJ2yBDOXfCLpaaf1VctqJ93KhONDLPnM2pREK48/fM0mn6x9uuCSJXo3FSW9xjnUXzj7cGxOvtdX2qnfTitiIiPIZNb4k6qbBbft1rI8oHYnt5+YUeVcrsSpoLL2QNAx9bavhv/hTO6Av/7Qofw/caGcSpx9kPGaxMjSBIpa32zrwRj/0qDVm7xgp1qkZayv54URd/nyqrNx0ZC8ZxS2mKLGWiL5bsOkNcEGe0lzeUCAwEAAQ==" 
```
(note the `" "`s in the middle, and the space after the trailing `"`). For handling this correctly, I think the module should strip quoting from the API result (and from the module input, in case someone passes in quoted TXT entries).

**Comment by felixfontein** [comment]
Another experience: when I set a short enough TXT entry (i.e. <= 256 bytes) without spaces, Hetzner's API keeps the quotes if I add them, and doesn't add them if I don't add them. So if I pass `test` without quotes it is returned without quotes, and if I pass `"test"` with quotes it is returned with quotes.

Maybe there should be an option which tells the module how to handle quotes in TXT entries? So that people who want to explicitly set the quotes the way they want can do it however they want, and others (probably essentially everyone) can just let the module take care of the quotes.

**Comment by felixfontein** [comment]
(I guess this should happen in a follow-up PR though...)

**Pull request merged**